File: Backend/Events/Converters/Parties/Digimons/VitalsConverter.cs
using Backend.Domain.Models.Parties.Digimons;
using Backend.Events.DTO.Parties.Digimons;

namespace Backend.Events.Converters.Parties.Digimons;

public static class VitalsConverter
{
    public static VitalsDTO ToDTO(Vitals vitals) => new()
    {
        MaxHP = vitals.MaxHP,
        MaxMP = vitals.MaxMP,
        CurrentHP = vitals.CurrentHP,
        CurrentMP = vitals.CurrentMP
    };
}

File: Frontend/src/events/converters/parties/digimons/vitals.converter.ts
import type { VitalsDTO } from "@/events/dto/parties/digimons/vitals.dto";
import type { Vitals } from "@/models/party/digimon/vitals";

export class VitalsConverter {
    public static convert(newVitalsDto: VitalsDTO | null): Vitals {
        return {
            currentHP: newVitalsDto?.currentHP ?? 0,
            maxHP: newVitalsDto?.maxHP ?? 0,
            currentMP: newVitalsDto?.currentMP ?? 0,
            maxMP: newVitalsDto?.maxMP ?? 0
        };
    }
}